client/src/components/Board.js

```diff
@@ -39,6 +40,7 @@ class Board extends Component {
       if (selection === null) {
         this.setState({ message: 'Invalid selection' });
       } else {
+        this.state.originDestCoord.push([x, y]);
```

Comment:
@ShawnShiFeng You can't change state directly like this.  You need to use this.setState